src/api/vdom.js

```diff
@@ -295,6 +295,21 @@ function newElementVoid (tag, ...args) {
 // Text override ensures their calls can queue if using function helpers.
 const newText = wrapIdomFunc(text);
 
+export function child (ch) {
+  const ctype = typeof ch;
+  if (ctype === 'function') {
+    ch();
+  } else if (ctype === 'string' || ctype === 'number') {
+    newText(ch);
+  } else if (Array.isArray(ch)) {
+    ch.forEach(child);
+  } else if (ctype === 'object' && ch.$$typeof) {
+    const { key, props, ref, type } = ch;
```

Comment:
I had my concerns, but it really hit me when I was doing something simple in React (not at work) and wanted to pass off a node to a custom element table (that can't have slots because `<tr>` can't be a child of a custom element, etc). I wanted to use Storybook, and opted to just use react for the main story parts to render out the custom elements within the storybook. No can do.

I also think this is a major step in supporting framework interop. I'm not sure exactly how we'd do this as a separate package because we'd need to expose hooks for it. I think in some way, shape or form, we need to support the large frameworks in some way, and make it simple for consumers to use (like not have to search for and find new libs). I was actually caught off guard that this *didn't* work until I figured out that I was passing react nodes off to iDOM. Passing props and it "just working" is something that's easy to take for granted. I think others would be surprised, even though it makes sense when you think about it.

This probably also works with Preact (cc @developit). I'd also like to support real DOM nodes. All of these things are things that https://github.com/skatejs/dom-diff already supports, along with web workers.

If we made this more generic and removed the `$$typeof` check, then this actually would be mostly agnostic to react nodes and it would be just accepting an object with a particular schema, or schemas if we supported more reactalike libs.

Open to suggestions about how to do this in a more decoupled manner, but I think this is a pretty rad feature that would make writing web components within other virtual dom frameworks a breeze. It would definitely one-up other offerings.